R/pkg/R/mllib_classification.R

```diff
@@ -96,7 +98,8 @@ setClass("NaiveBayesModel", representation(jobj = "jobj"))
 #' @note spark.svmLinear since 2.2.0
 setMethod("spark.svmLinear", signature(data = "SparkDataFrame", formula = "formula"),
           function(data, formula, regParam = 0.0, maxIter = 100, tol = 1E-6, standardization = TRUE,
-                   threshold = 0.0, weightCol = NULL, aggregationDepth = 2) {
+                   threshold = 0.0, weightCol = NULL, aggregationDepth = 2, solver = "l-bfgs",
```

Comment:
for parameter with limited choice,
use `c("l-bfgs", "owlqn")` https://github.com/apache/spark/pull/18140/files#diff-2e1e5d49cfda9702645a955f584572eeR129
and match.arg instead https://github.com/apache/spark/pull/18140/files#diff-2e1e5d49cfda9702645a955f584572eeR132